### PR TITLE
Gpg 867 1 elastic beanstalk

### PR DESCRIPTION
**Pull request by LadunOmideyiSoftwire**
This contains:

- The vpc and elastic beanstalk application
- The manifest json file  _NOTE: the Build Gender Pay Gap - AWS pipeline won't succeed until this is merged because it's necessary (along with the archive file) for azure devops to deploy elastic beanstalk_. 
- Environment specific backends
- Additional providers

### CODE REVIEW

File: terraform/elasticbeanstalk.tf [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+// Archive file 
+data "aws_s3_object" "gpg-archive-zip" {
+  bucket = data.aws_s3_bucket.gpg-application-version-storage.id
+  key    = "publish.zip"
+}
+
+// Application
+resource "aws_elastic_beanstalk_application" "gpg-application" {
+  name        = join("-",["gpg-application", var-env])
```
Comment (davidm-m) [inline terraform/elasticbeanstalk.tf:14]: Is this supposed to be `var-env` or `var.env`?

File: terraform/dev-app.tfvars [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿Instance_type          = "t2.medium"
```
Comment (davidm-m) [inline terraform/dev-app.tfvars:1]: Are we likely to actually need a medium here? Have we looked into whether we could get away with a small on dev and pre-prod?

File: terraform/main.tf [hunk 3]
```diff
@@ -1,15 +1,6 @@
...
-  bucket = "gpg-bucket-${terraform.workspace}"
-  acl    = "private"
-
-  versioning = {
-    enabled = true
-  }
-
+data "aws_availability_zones" "available" {
+  state = "available"
 }
```
Comment (davidm-m) [inline terraform/main.tf:6]: Showing how little I know about terraform here, but do we need to do more than this in the `main.tf` file? Or does it just detect there are more `.tf` files and run them all?

**Review by davidm-m — CHANGES_REQUESTED** [review]
A typo and a couple of questions. Overall looks very good, the separation of types of resources into multiple files is really nice!

File: terraform/elasticbeanstalk.tf (hunk 1 above)
Comment (LadunOmideyiSoftwire) [reply terraform/elasticbeanstalk.tf:14]: Definitely the latter!

File: terraform/main.tf (hunk 3 above)
Comment (LadunOmideyiSoftwire) [reply terraform/main.tf:6]: It detects all .tf files in the root terraform folder so we should be fine.

File: terraform/dev-app.tfvars (hunk 2 above)
Comment (LadunOmideyiSoftwire) [reply terraform/dev-app.tfvars:1]: We should be able to get away with a small and I've made the changes.

**Review by davidm-m — APPROVED** [review]
Nice!

**Pull request merged**